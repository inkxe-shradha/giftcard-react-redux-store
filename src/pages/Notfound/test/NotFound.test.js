import { createTheme, ThemeProvider } from "@mui/material";
import { mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import NotFound from "../NotFound";

describe("NotFound component suite case", () => {
  const theme = createTheme();
  const customRender = (props = {}) => {
    return mount(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <NotFound {...props} />
        </BrowserRouter>
      </ThemeProvider>
    );
  };
  it("Should render the component snapshot", () => {
    const view = customRender();
    expect(view.html()).toMatchSnapshot();
  });

  it("Should contain a anchor tag that would have '/' href value", () => {
    const view = customRender();
    expect(view.find("a").prop("href")).toBe("/");
  });
});
