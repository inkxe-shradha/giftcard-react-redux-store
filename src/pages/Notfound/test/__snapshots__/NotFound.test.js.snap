// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component suite case Should render the component snapshot 1`] = `"<section class=\\"page_404\\"><div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-sm-12 \\"><div class=\\"col-sm-12 text-center\\"><div class=\\"four_zero_four_bg\\"><h1 class=\\"text-center \\">404</h1></div><div class=\\"contant_box_404\\"><h3 class=\\"h2\\">Look like you are lost</h3><p>the page you are looking for not available!</p><a class=\\"link_404\\" href=\\"/\\">Go to Home</a></div></div></div></div></div></section>"`;
